### PR TITLE
feat(headers): internal HTTP headers refactoring

### PR DESCRIPTION
**Pull request by ytvnr**
**Issue**

https://github.com/gravitee-io/issues/issues/7025

**Description**

Update `gravitee-gateway-api` and `gravitee-common` dependencies to benefit from headers refactoring

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.13.0-SNAPSHOT.issues-7025-resttosoap-headers`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-rest-to-soap/1.13.0-SNAPSHOT.issues-7025-resttosoap-headers/gravitee-policy-rest-to-soap-1.13.0-SNAPSHOT.issues-7025-resttosoap-headers.zip)

### CODE REVIEW

**Review by brasseld — APPROVED** [review]

**Review by phiz71 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -35,9 +35,9 @@
     </parent>
 
     <properties>
-        <gravitee-gateway-api.version>1.22.0</gravitee-gateway-api.version>
+        <gravitee-gateway-api.version>1.31.0</gravitee-gateway-api.version>
         <gravitee-policy-api.version>1.10.0</gravitee-policy-api.version>
-        <gravitee-common.version>1.18.0</gravitee-common.version>
+        <gravitee-common.version>1.24.0</gravitee-common.version>
```
Comment (gaetanmaisse) [inline pom.xml:40]: Should be `1.25.0` no?

**Review by gaetanmaisse — COMMENTED** [review]
Not sure to see how it will fix the issue as these deps are `provided` so bumping them doesn't change the final code

**Review by gaetanmaisse — APPROVED** [review]
All's good 🚀

**Pull request merged**